src/ralph/data_center/admin_actions.py

```diff
@@ -0,0 +1,81 @@
+import re
+from typing import Union
+
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.data_center.models import DataCenterAsset
+
+
+def assign_management_hostname_and_ip(modeladmin, request, queryset):
+    for dca in queryset:
+        try:
+            if not modeladmin.has_change_permission(request, obj=dca):
+                raise RuntimeError("insufficient permissions")
```

Comment:
Because of mess with actions this action is visible for everyone so I guard here